app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    def initialize(account, payment_status, user_session = nil)
+      @account            = account
+      @payment_successful = payment_status
+      @gateway_setting    = account.gateway_setting
+      @user_session       = user_session
+    end
+
+    def call
+      return if payment_successful
+
+      limiter = ActionLimiter.new(@account)
```

Comment:
```suggestion
      limiter = ActionLimiter.new(account)
```

Since you created the reader method anyway :smile: